src/ui/constants/visualizations/metric.ts

```diff
@@ -16,7 +17,14 @@ export const METRIC_VISUALIZATION: Shared['visualization'] = {
         {
             allowedTypes: ITEM_TYPES.DIMENSIONS_AND_MEASURES,
             allowedFinalTypes: ITEM_TYPES.MEASURES,
-            allowedDataTypes: PRIMITIVE_DATA_TYPES,
+            allowedDataTypes: PRIMITIVE_DATA_TYPES_AND_MARKUP,
+            checkAllowed: (item: Field) => {
+                const allowedDataTypes = Utils.isEnabledFeature(Feature.MarkupMetric)
+                    ? PRIMITIVE_DATA_TYPES_AND_MARKUP
+                    : PRIMITIVE_DATA_TYPES;
+
+                return allowedDataTypes.has(item.data_type);
+            },
```

Comment:
It's a bit confusing - why are there the two fields with different types?  I think we should leave only one of the two